packages/core/upload/admin/src/components/TableList/index.js

```diff
@@ -1,32 +1,49 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { useIntl } from 'react-intl';
+import { useTracking } from '@strapi/helper-plugin';
 import { BaseCheckbox } from '@strapi/design-system/BaseCheckbox';
 import { Table, Th, Thead, Tr } from '@strapi/design-system/Table';
 import { Typography } from '@strapi/design-system/Typography';
 import { VisuallyHidden } from '@strapi/design-system/VisuallyHidden';
 
 import { getTrad } from '../../utils';
-import { AssetDefinition, tableHeaders } from '../../constants';
+import { AssetDefinition, tableHeaders, FolderDefinition } from '../../constants';
 import { TableRows } from './TableRows';
 
-export const AssetTableList = ({ assets, onEditAsset, onSelectAsset, selectedAssets }) => {
+export const TableList = ({
+  assetCount,
+  folderCount,
+  onEditAsset,
+  onEditFolder,
+  onSelectAll,
+  onSelectOne,
+  rows,
+  selected,
+}) => {
   const { formatMessage } = useIntl();
+  const { trackUsage } = useTracking();
 
   return (
-    <Table colCount={tableHeaders.length} rowCount={assets.length}>
+    <Table colCount={tableHeaders.length + 2} rowCount={assetCount + folderCount + 1}>
       <Thead>
         <Tr>
           <Th>
-            {/* TODO: Replace ML BaseCheckbox select all with this one when adding folders to Table */}
             <BaseCheckbox
               aria-label={formatMessage({
                 id: getTrad('bulk.select.label'),
                 defaultMessage: 'Select all folders & assets',
               })}
-              onValueChange={() => {}}
-              // TODO: remove disabled once this checkbox will replace the current one used to bulk select
-              disabled
+              indeterminate={selected?.length > 0 && selected?.length !== assetCount + folderCount}
+              onChange={(e) => {
```

Comment:
I think this should be in it's own handler imo

Reply:
you've resolved it but it's not been handled yet?

Reply:
I just realized I thought you were talking about the `indeterminate` part that was repeated in different places, so I fixed that but not the `onChange` part 😅